app/api/users/api.py

```diff
@@ -62,17 +80,44 @@ async def create_user(data: CreateUserData, user_api: OptScaleUserAPI = Depends(
         JWTBearer: Ensures that the request is authenticated using a valid JWT.
     """
     try:
+        auto_verified, check_user_email = await process_authentication_info(jwt_payload)
```

Comment:
I would suggest the following:

Start initializing `verify_email = True`

if you don't receive a token from the JWTBearer dependency you check using the `OptScaleInvitationAPI` client if there is an invitation for such email. If there isn't ,you must return a 401. If the invitation exists, in that case set `verify_email = False`.

Then you can proceed with user creation.

This avoid using the invitation api client from the user api client.